### PR TITLE
Add installation support for private cluster on aws

### PR DESCRIPTION
**Pull request by mhanss**

### CODE REVIEW

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml [hunk 1]
```diff
@@ -6,7 +6,6 @@ base_images:
 binary_build_commands: make build
 build_root:
   from_repository: true
-  use_build_cache: true
```
Comment (jmguzik) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml:9]: @petr-muller please comment on that change.

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.9.yaml [hunk 2]
```diff
@@ -6,7 +6,6 @@ base_images:
 binary_build_commands: make build
 build_root:
   from_repository: true
-  use_build_cache: true
```
Comment (jmguzik) [inline ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.9.yaml:9]: @petr-muller please comment on that change.

**Comment by jmguzik** [comment]
/cc @petr-muller From slack Conv I had with Manoj:

> I found an issue with windows-machine-config-operator-release-4.12 and 4.9in unified CI. It seems windows-machine-config-operator is failing due to use_build_cache: true .After removing build_cache it starts working.

Could you consult on that?

File: ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml (hunk 1 above)
Comment (petr-muller) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml:9]: Comment in what sense? I need at least some context

Comment (jmguzik) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml:9]: sorry, please see the last message in the thread.

Comment (jmguzik) [reply ci-operator/config/openshift/windows-machine-config-operator/openshift-windows-machine-config-operator-release-4.12.yaml:9]: https://github.com/openshift/release/pull/26414#issuecomment-1058008871

**Comment by petr-muller** [comment]
Hard to consult anything without evidence,symptoms or artifacts. If the feature breaks something then it could be a bug, a flake or some weird corner case. Ideally we'd get a proper bug report about the failure.

Removing the option is completely safe, the option is supposed to be just an optimization. It's documented here: https://docs.ci.openshift.org/docs/architecture/ci-operator/#using-the-build-cache - the cached build_root is the `bin` image promoted during the previous promotion run - it contains the go cache from the previous built, so when you use it as a build root, you get the previous go cache and do not rebuild code that did not change.

**Comment by mhanss** [comment]
> Hard to consult anything without evidence,symptoms or artifacts. If the feature breaks something then it could be a bug, a flake or some weird corner case. Ideally we'd get a proper bug report about the failure.
> 
> Removing the option is completely safe, the option is supposed to be just an optimization. It's documented here: https://docs.ci.openshift.org/docs/architecture/ci-operator/#using-the-build-cache - the cached build_root is the `bin` image promoted during the previous promotion run - it contains the go cache from the previous built, so when you use it as a build root, you get the previous go cache and do not rebuild code that did not change.

@petr-muller agreed, yes this option is for optimization but it is failing with this error `error="failed to ensure imagestreamtags in cluster build03: failed waiting for imagestreamtag build-cache/openshift-windows-machine-config-operator:release-4.9 to appear`. Here is the [logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/26414/pull-ci-openshift-release-master-pj-rehearse/1499019236469641216) of failed job.

**Comment by petr-muller** [comment]
@mhanss Oh, but this is something different. This is rehearsals failing to set up the cache (which *is* technically a problem but I do not know why would you care, this should not affect anything relevant to you). This simply means that the rehearsals run without the cached build root, which should be fine (do you observe something else?), and it also should not affect the jobs when they run normally (not as rehearsals).

**Comment by mhanss** [comment]
@petr-muller thanks for the clarification. Should I reverse the changes?

**Comment by petr-muller** [comment]
That depends on you, but unless you see actual problems with running the jobs, I'd suggest to keep using the cache.

**Comment by mhanss** [comment]
Closing this as it will be covered as a part of https://github.com/openshift/release/pull/22260

**Pull request closed without merging**